src/matrix_content_scanner/httpserver.py

```diff
@@ -26,7 +26,44 @@
 
 logger = logging.getLogger(__name__)
 
-_media_path_regexp = r"/{media_path:.+}"
+_MEDIA_PATH_REGEXP = r"/{media_path:.+}"
+
+_CORS_HEADERS = {
+    "Access-Control-Allow-Origin": "*",
+    "Access-Control-Allow-Methods": "GET, POST, OPTIONS",
+    "Access-Control-Allow-Headers": "Origin, X-Requested-With, Content-Type, Accept, Authorization",
+}
+
+
+@web.middleware
+async def simple_cors_middleware(
+    request: web.Request,
+    handler: Callable[[web.Request], Awaitable[web.Response]],
+) -> web.Response:
+    """A simple aiohttp middleware that adds CORS headers to responses, and handles
+    OPTIONS requests.
+
+    Args:
+        request: The request to handle.
+        handler: The handler for this request.
+
+    Returns:
+        A response with CORS headers.
+    """
+    if request.method == "OPTIONS":
+        # We don't register routes for OPTIONS requests, therefore the handler we're given
+        # in this case just raises a 405 Method Not Allowed status using an exception.
+        # Because we actually want to return a 200 OK with additional headers, we ignore
+        # the handler and just return a new response.
+        response = web.Response(
+            headers=_CORS_HEADERS
+        )
+        return response
```

Comment:
Does this default to 200 OK? Looks like it from https://docs.aiohttp.org/en/stable/web_reference.html#response

Reply:
Yep, I'll make it explicit though